src/binding/c/attr_api.txt

```diff
@@ -0,0 +1,229 @@
+    .skip: Fortran
+/*
+    Notes:
+        Attributes must be extracted from the same language as they were inserted
+        in with 'MPI_ATTR_PUT'.  The notes for C and Fortran below explain why.
+*/
+/*
+    Notes for C:
+        Even though the 'attribute_val' argument is declared as 'void *', it is
+        really the address of a void pointer (i.e., a 'void **').  Using
+        a 'void *', however, is more in keeping with C idiom and allows the
+        pointer to be passed without additional casts.
+
+    .N Fortran
+        The 'attribute_val' in Fortran is a pointer to a Fortran integer, not
+        a pointer to a 'void *'.
+*/
+
+MPI_Type_set_attr:
+    .desc: Stores attribute value associated with a key
+    .skip: ThreadSafe
+    .polymorph: MPIR_Attr_type attr_type=MPIR_ATTR_PTR
+/*
+    Notes:
+
+    The type of the attribute value depends on whether C or Fortran is being used.
+    In C, an attribute value is a pointer ('void *'); in Fortran, it is an
+    address-sized integer.
+
+    If an attribute is already present, the delete function (specified when the
+    corresponding keyval was created) will be called.
+*/
+
+MPI_Win_create_keyval:
+    .desc: Create an attribute keyval for MPI window objects
+    .extra: AttrErrReturn
+/*
+    Notes:
+    Key values are global (available for any and all communicators).
+
+    Default copy and delete functions are available.  These are
+    + MPI_WIN_NULL_COPY_FN   - empty copy function
+    . MPI_WIN_NULL_DELETE_FN - empty delete function
+    - MPI_WIN_DUP_FN         - simple dup function
+
+    There are subtle differences between C and Fortran that require that the
+    copy_fn be written in the same language from which 'MPI_Win_create_keyval'
+    is called.
+    This should not be a problem for most users; only programmers using both
+    Fortran and C in the same program need to be sure that they follow this rule.
+*/
+
+MPI_Win_delete_attr:
+    .desc: Deletes an attribute value associated with a key on a window
+
+MPI_Win_free_keyval:
+    .desc: Frees an attribute key for MPI RMA windows
+{
+    MPIR_free_keyval(win_keyval_ptr);
+    *win_keyval = MPI_KEYVAL_INVALID;
+}
+
+MPI_Win_get_attr:
+    .desc: Get attribute cached on an MPI window object
+    .polymorph: MPIR_Attr_type attr_type=MPIR_ATTR_PTR
+    .skip: Fortran
+/*
+    Notes:
+        Attributes must be extracted from the same language as they were inserted
+        in with 'MPI_ATTR_PUT'.  The notes for C and Fortran below explain why.
+*/
+/*
+    Notes for C:
+        Even though the 'attribute_val' argument is declared as 'void *', it is
+        really the address of a void pointer (i.e., a 'void **').  Using
+        a 'void *', however, is more in keeping with C idiom and allows the
+        pointer to be passed without additional casts.
+
+    .N Fortran
+        The 'attribute_val' in Fortran is a pointer to a Fortran integer, not
+        a pointer to a 'void *'.
```

Comment:
Do we need to add this comment?
https://github.com/pmodels/mpich/blob/main/src/mpi/attr/win_get_attr.c#L108-L113

Reply:
Will add.

Reply:
Added.